mango_time_series/mango_time_series/dashboards/time_series_utils/ui_components.py

```diff
@@ -0,0 +1,1008 @@
+                showlegend=False,
+                height=900,
+                width=800,
+            )
+
+            st.plotly_chart(fig, use_container_width=True)
+
+    # "Seasonality boxplot"
+    elif select_plot == UI_TEXT["plot_options"][4]:
+        selected_granularity = select_agr_tmp
+
+        if selected_granularity == UI_TEXT["daily"]:
+            freq_options = UI_TEXT["frequency_options"]
+        elif selected_granularity == UI_TEXT["weekly"]:
+            freq_options = UI_TEXT["frequency_options"][1:]
+        elif selected_granularity == UI_TEXT["monthly"]:
+            freq_options = [UI_TEXT["frequency_options"][2]]
+        else:
+            st.write(UI_TEXT["boxplot_error"])
+            return
+        selected_freq = st.selectbox(UI_TEXT["select_frequency"], freq_options)
+
+        for serie in selected_series:
+            selected_data = time_series.copy()
+            filter_cond = ""
+            for col, col_value in serie.items():
+                filter_cond += f"{col} == '{col_value}' & "
+
+            # Remove the last & from the filter condition
+            filter_cond = filter_cond[:-3]
+            if filter_cond:
+                selected_data = selected_data.query(filter_cond)
+
+            selected_data = selected_data.set_index("datetime")
+            selected_data.index = pd.to_datetime(selected_data.index)
+
+            if selected_freq == UI_TEXT["frequency_options"][0]:
+                selected_data["day_of_year"] = selected_data.index.dayofyear
+                fig = px.box(
+                    selected_data,
+                    x="day_of_year",
+                    y="y",
+                    title=UI_TEXT["boxplot_titles"]["daily"],
+                )
+                st.plotly_chart(fig, use_container_width=True)
+
+            elif selected_freq == UI_TEXT["frequency_options"][1]:
+                selected_data["day_of_week"] = selected_data.index.weekday
+                fig = px.box(
+                    selected_data,
+                    x="day_of_week",
+                    y="y",
+                    title=UI_TEXT["boxplot_titles"]["weekly"],
+                )
+                st.plotly_chart(fig, use_container_width=True)
+
+            elif selected_freq == UI_TEXT["frequency_options"][2]:
+                selected_data["month"] = selected_data.index.month
+                fig = px.box(
+                    selected_data,
+                    x="month",
+                    y="y",
+                    title=UI_TEXT["boxplot_titles"]["monthly"],
+                )
+                st.plotly_chart(fig, use_container_width=True)
+
+    # "Periodogram"
+    elif select_plot == UI_TEXT["plot_options"][5]:
+        if not cast_env_to_bool("ENABLE_EXPERIMENTAL_FEATURES", default=False):
+            st.warning(UI_TEXT["experimental_features_warning"], icon="⚠️")
+            return
+        else:
+            st.info(UI_TEXT["experimental_features_info"], icon="ℹ️")
```

Comment:
Do we actually have emojis on a .py file?

Reply:
Streamlit admit icons but i will change to shortcodes

Reply:
https://streamlit-emoji-shortcodes-streamlit-app-gwckff.streamlit.app/

Reply:
If i do it i get this exception: StreamlitAPIException: The value "ℹ️" is not a valid emoji. Shortcodes are not allowed, please use a single character instead.